### ISSUE TITLE
Regression in DVR and Isosurface on Mac

### ISSUE THREAD

**Issue by clyne** [NCAR/VAPOR#2265]
A regression has been introduced on the Mac (and possibly elsewhere) in the regular grid algorithm for both DVR and Isosurface rendering. See attached image. The curvilinear grid algorithm works just fine. 

**Helpful additional information**
(*Please click check boxes AFTER submitting ticket*)
- [ ] Did VAPOR crash?
- [x] Did you get wrong results?

**Impact**
- [x] High - User productivity significantly degraded
- [ ] Medium - User productivity partially degraded
- [ ] Low - User productivity trivially degraded

### To Reproduce

1. Load Katrina.
2. Create and enable the DVR or Iso renderer 

*Additionally, please attach a session file to this ticket to easily reproduce (use File->Save Session).*

### Desktop
  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro14,2
  Processor Name:	Intel Core i7
  Processor Speed:	3.5 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache (per Core):	256 KB
  L3 Cache:	4 MB
  Hyper-Threading Technology:	Enabled
  Memory:	16 GB
  Boot ROM Version:	198.71.1.0.0
  SMC Version (system):	2.44f1
  Serial Number (system):	C02V8137HV2T
  Hardware UUID:	F2EA4D07-2D45-5644-9B4B-3C33D11CF2F5
  Mac OSX 10.14.6

![DVRBroken](https://user-images.githubusercontent.com/6777736/78267474-421f9700-74c4-11ea-9e46-b59767932fa6.png)

**Comment by StasJ** [comment]
@clyne I cannot reproduce this:

![katrina](https://user-images.githubusercontent.com/2772687/78277249-6df54980-74d1-11ea-97dd-2d33ea2422b5.png)

**Comment by clyne** [comment]
@StasJ any thoughts on how we can troubleshoot it? My system is only two years old and a pretty common platform. I don't think we can ignore it.

**Comment by StasJ** [comment]
@clyne I agree. The first step may be to make sure we are using the same version/timestep of Katrina. Below are my md5 checksums.

```
b8a6c987868de78956493825478c321c  Katrina-3.1.0.nc
f29a42d04f7f1f35663fa69151f0d8c4  Katrina-3.1.0_data
```

Next maybe you could check to see if the same problem persists in vapor 3.0. The regular grid renderer has not changed much. If that helps we can try to figure out at which commit this was introduced.

From the artifacts and the fact that it is localized to your machine it is possible that this bug is occurring within the driver or hardware on your machine. So if the above does not help, we may need to schedule a screen share and we can look at it together on your machine.

**Comment by clyne** [comment]
> @clyne I agree. The first step may be to make sure we are using the same version/timestep of Katrina. Below are my md5 checksums.
> 
> ```
> b8a6c987868de78956493825478c321c  Katrina-3.1.0.nc
> f29a42d04f7f1f35663fa69151f0d8c4  Katrina-3.1.0_data
> ```
> 

I should have been more clear. It happens with any data set.

> Next maybe you could check to see if the same problem persists in vapor 3.0. The regular grid renderer has not changed much. If that helps we can try to figure out at which commit this was introduced.
> 

Good idea. I ran the 3.2.0 release and it doesn't exhibit the same behavior, but there are artifacts that have always been present (see attachment). The difference now is that the current build is completely unusable.

> From the artifacts and the fact that it is localized to your machine it is possible that this bug is occurring within the driver or hardware on your machine. So if the above does not help, we may need to schedule a screen share and we can look at it together on your machine.

How do we know that it is localized to my machine? We only have two data points as far as I know.


![DVRBrokenVAPOR320](https://user-images.githubusercontent.com/6777736/78304740-a2cac600-74fc-11ea-8416-0b5df8488b08.png)

**Comment by StasJ** [comment]
Ah now I remember there was always a problem with your machine. This is definitely a driver/hardware problem then. I am assuming it is localized to your machine since IIRC it has had these problems since 3.0 and we have not yet found any other machine (on any OS) that has such problems as far as I am aware.

I can debug this further but I need access to your machine. It may be practical if you created a new non-privileged user that I can use without having access any of your files. Then we can set up a time when you are not working and I can remote access your machine.

Otherwise you could try figuring out what commit worsened the problem and we can take it from there.

**Comment by clyne** [comment]
We're going to table this until we're back on the Mesa and no longer under stay at home orders for corona virus.

**Comment by clyne** [comment]
Re-opening this issue. It is also occurring on my new Mac OSX 11.4 system. The one significant common element between my old and new Mac is that they both have the same graphics adapter: Intel Iris Plus Graphics

**Comment by sgpearse** [comment]
Since mac is moving to M1 architecture, intel is being phased out, so our consensus is tot not fix it.
